### ISSUE TITLE
update() breaks when sequelize configured with raw:true

### ISSUE THREAD

**Issue by buske** [feathersjs/feathers-sequelize#99]
Affects master, v1.4.3, etc.

A call to `update()` results in a `Model.findById` call:
https://github.com/feathersjs/feathers-sequelize/blob/master/src/index.js#L168

If sequelize is configured with `raw: false` (the default), this returns a DAO, with a `toJSON` method, which is then called several lines later:
https://github.com/feathersjs/feathers-sequelize/blob/master/src/index.js#L174

However, if sequelize is configured with `raw: true`, the `findById` call returns a JSON object without this method.

There are two simple fixes:
1. Only call `toJSON` if it exists and is a function
1. pass `{raw: false}` as a second argument to the `findById` call to force the expected behaviour

**Comment by daffl** [comment]
Makes sense. This can probably easily be fixed by changing the line you pointed out to

```js
Object.keys(typeof instance.toJSON === 'function' ? instance.toJSON() : instance).forEach(key => {
```

**Comment by daffl** [comment]
Thanks to @lowip this has been fixed in #100 and released as v1.4.4

**Comment by lowip** [comment]
Unfortunately my PR might have been too fast

In order to update, the DOA instance is used:
https://github.com/feathersjs/feathers-sequelize/blob/master/src/index.js#L182
```javascript
return instance.update(copy, options);
```

**Comment by lowip** [comment]
The best solution I see would be replacing:
https://github.com/feathersjs/feathers-sequelize/blob/master/src/index.js#L168
```javascript
return this.Model.findById(id).then(instance => {
```
with
```javascript
return this.Model.findById(id, { raw: false }).then(instance => {
```

**Comment by daffl** [comment]
Ah, I see. Well at least we know it didn't break anything else ;)

I think we should do what `patch` does and use `this.Model.update(omit(data, this.id), options)` instead of calling `save` on the model instance.

**Comment by lowip** [comment]
I updated the PR #101, would love some feedback on the issue I'm raising here:
https://github.com/feathersjs/feathers-sequelize/pull/101/commits/b6a4128a2e9120aaa3cef755621a498cdc84833b

Thanks!

**Comment by daffl** [comment]
Now it's been fixed via #101 in v1.4.5
